crates/http-source/src/main.rs

```diff
@@ -13,29 +15,39 @@ use fluvio_connector_common::{
     Source,
 };
 
-use http_streaming_source::HttpStreamingSource;
+use crate::http_streaming_source::reconnect_stream_with_backoff;
 use source::HttpSource;
 
 const SIGNATURES: &str = concat!("InfinyOn HTTP Source Connector ", env!("CARGO_PKG_VERSION"));
+const BACKOFF_LIMIT: u64 = 100000; //1000 seconds
 
+#[allow(unreachable_code)]
 #[connector(source)]
 async fn start(config: HttpConfig, producer: TopicProducer) -> Result<()> {
     debug!(?config);
-
-    let mut stream = if config.stream {
-        HttpStreamingSource::new(&config)?.connect(None).await?
-    } else {
-        HttpSource::new(&config)?.connect(None).await?
-    };
-
-    info!("Starting {SIGNATURES}");
-
-    while let Some(item) = stream.next().await {
-        trace!(?item);
-        producer.send(RecordKey::NULL, item).await?;
+    let mut backoff = Backoff::new();
+
+    loop {
+        let mut stream = if config.stream {
+            match reconnect_stream_with_backoff(&config, &mut backoff).await {
+                Ok(stream) => stream,
+                Err(_err) => {
+                    continue;
```

Comment:
Not sure the best place to put it, but there should be a way for a user to see and assess reconnects in the regular logs.  I think a `warn` or `error` log output on disconnect, and an `info` on successful connect would be sufficient for that.